### PR TITLE
[Reland] Enable XPUEvent elapsed_time function

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140873
* #140865

# Motivation
This PR intends to reland https://github.com/pytorch/pytorch/pull/134666 that has been reverted in https://github.com/pytorch/pytorch/pull/140872
We reverted it because I forgot to support `elapsed_time` for `XPUGuardImpl`, which resulted in `c10::Event` not supporting' elapsed_time' and blocking XPU CI. 

# Additional Context
We split https://github.com/pytorch/pytorch/pull/134666 into two parts: one part, PR #140865, supports `elapsed_time` for `torch.Event` and another one, this PR, supports for `torch.xpu.elapsed_time`.

cc @gujinghui @EikanWang @fengyuan14

### CODE REVIEW

File: test/test_xpu.py [hunk 1]
```diff
@@ -235,6 +236,21 @@ def test_events(self):
         stream.record_event(event)
         event.synchronize()
         self.assertTrue(event.query())
+        start_event = torch.xpu.Event(enable_timing=True)
+        end_event = torch.xpu.Event(enable_timing=True)
+        stream.record_event(start_event)
+        time.sleep(0.1)
+        stream.record_event(end_event)
+        torch.xpu.synchronize()
+        if int(torch.version.xpu) >= 20250000:
```
Comment (gujinghui) [inline test/test_xpu.py:245]: It's not related to this PR. But why torch.version.xpu is going to tell the version of compiler?

Comment (guangyey) [reply test/test_xpu.py:245]: Aligning to cuda version, like 11.8, 12.4...

**Review by gujinghui — APPROVED** [review]

**Comment by guangyey** [comment]
"Unrelated failures, there is an issue to track it https://github.com/pytorch/pytorch/issues/141705"
@pytorchbot merge -i

**Pull request closed without merging**